### PR TITLE
PCAP: Add raw SSL packets logging support

### PR DESCRIPTION
**Pull request by sepalani**
This PR allows the NetworkLogger to dump raw SSL packets in PCAP format when dumping decrypted SSL packets is disabled.

That way, most network related features would be supported by the logger except for WiiConnect24, DNS and ICMP packets.

Ready to be reviewed & merged.

Depends on https://github.com/dolphin-emu/dolphin/pull/9408.

### CODE REVIEW

**Comment by leoetlino** [comment]
Now that #9408 has been merged, this PR needs a rebase.

**Comment by sepalani** [comment]
@leoetlino 
Done.

File: Source/Core/Core/IOS/Network/SSL.cpp [hunk 1]
```diff
@@ -54,17 +55,30 @@ namespace
...
 {
   auto* ssl = static_cast<WII_SSL*>(ctx);
+  auto* fd = &ssl->hostfd;
 
   if (ssl->ctx.state == MBEDTLS_SSL_SERVER_HELLO)
     mbedtls_ssl_set_hostname(&ssl->ctx, ssl->hostname.c_str());
-  return mbedtls_net_send(&ssl->hostfd, buf, len);
+  const int ret = mbedtls_net_send(fd, buf, len);
+
+  // Log raw SSL packets if we don't dump unencrypted SSL write
```
Comment (leoetlino) [inline Source/Core/Core/IOS/Network/SSL.cpp:64]:
```suggestion
  // Log raw SSL packets if we don't dump unencrypted SSL writes
```

File: Source/Core/Core/IOS/Network/SSL.cpp [hunk 2]
```diff
@@ -54,17 +55,30 @@ namespace
...
+  return ret;
 }
 
 int SSLRecv(void* ctx, unsigned char* buf, size_t len)
 {
   auto* ssl = static_cast<WII_SSL*>(ctx);
+  auto* fd = &ssl->hostfd;
+  const int ret = mbedtls_net_recv(fd, buf, len);
+
+  // Log raw SSL packets if we don't dump unencrypted SSL read
```
Comment (leoetlino) [inline Source/Core/Core/IOS/Network/SSL.cpp:77]:
```suggestion
  // Log raw SSL packets if we don't dump unencrypted SSL reads
```

**Review by leoetlino — CHANGES_REQUESTED** [review]

**Comment by sepalani** [comment]
@leoetlino 
Done.

**Review by leoetlino — APPROVED** [review]

**Pull request merged**